FineTuning_v2.ipynb

```diff
@@ -303,41 +274,12 @@
     "shifted_ts = shifted_non_threads.lag(1).col(\"ts\").first(window=kd.windows.Since(is_new))\n",
     "thread_ts = ts.if_(is_new).else_(shifted_ts)\n",
     "\n",
-    "# replace the `thread_ts` in the data with our generated value and filter out messages we no longer need\n",
-    "non_threads_threads = non_threads.extend({\"thread_ts\": thread_ts}).filter(ts.is_not_null().and_(thread_ts.is_not_null()))\n",
-    "non_threads_threads.preview(5)"
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "id": "41dc1307",
-   "metadata": {},
-   "source": [
-    "#### 1.2.4 Rejoin all the messages into a single timestream\n",
-    "\n",
-    "Now that both threads and non-threads have the `threads_ts` set, we merge the data back into a single timestream."
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": null,
-   "id": "169b1d1d",
-   "metadata": {},
-   "outputs": [],
-   "source": [
-    "# Ideally we would just do the below, but there is a bug that currently prevents this from working\n",
-    "# joined = threads.else_(non_threads_threads)\n",
-    "\n",
-    "# Until the bug is fixed, this gets us the same result\n",
-    "joined = kd.record({\n",
-    "    \"ts\": threads.col(\"ts\").else_(non_threads_threads.col(\"ts\")),\n",
-    "    \"text\": threads.col(\"text\").else_(non_threads_threads.col(\"text\")),\n",
-    "    \"user\" : threads.col(\"user\").else_(non_threads_threads.col(\"user\")),\n",
-    "    \"thread_ts\" : threads.col(\"thread_ts\").else_(non_threads_threads.col(\"thread_ts\")),\n",
-    "    \"channel\" : threads.col(\"channel\").else_(non_threads_threads.col(\"channel\")),\n",
-    "})\n",
-    "\n",
-    "joined.preview(5)"
```

Comment:
this isn't fixed yet... we need to keep the above code: https://github.com/kaskada-ai/kaskada/issues/725